src/main/resources/static/templates/UnrestrictedFileUpload/LEVEL_1/FileUpload.js

```diff
@@ -1,13 +1,46 @@
+function genericResponseHandler2(xmlHttpRequest, callBack, isJson) {
+  if (xmlHttpRequest.readyState == XMLHttpRequest.DONE) {
+    if (xmlHttpRequest.status == 200 || xmlHttpRequest.status == 401) {
+      if (isJson) {
+        callBack(JSON.parse(xmlHttpRequest.responseText));
+      } else {
+        callBack(xmlHttpRequest.responseText);
+      }
+    } else if (xmlHttpRequest.status == 400) {
+      alert("There was an error 400");
+    } else if (xmlHttpRequest.status == 500) {
+      document.getElementById("uploaded_file_info").innerHTML =
+        "No space left on the device";
+    } else {
+      alert("something else other than 200/401 was returned");
+    }
+  }
+}
+
 function addingEventListenerToUploadImage() {
   document.getElementById("upload").addEventListener("click", function () {
-    var form = document.getElementById("file");
-    var file = form.files[0];
-    var formData = new FormData();
-    formData.append("file", file);
-    let url = getUrlForVulnerabilityLevel();
-    doPostAjaxCall(uploadImage, url, true, formData);
+    const form = document.getElementById("file");
+    const file = form.files[0];
+    if (file.size > 1048576) {
+      document.getElementById("uploaded_file_info").innerHTML =
+        "Max upload size exceeded exception : " +
```

Comment:
one more thing, i don't see any spring property update, does that means you it still blocks 10MB file?

Reply:
We didn't add any spring property because this will affect the other levels and the entire applications. We just added the following : test file.size > 1048576, in order to handle the exception of Maximum Size Configured.